src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.InFlightSolution.cs

```diff
@@ -117,8 +140,8 @@ public ImmutableArray<Task> DecrementInFlightCount_NoLock()
                 if (InFlightCount != 0)
                     return ImmutableArray<Task>.Empty;
 
-                _cancellationTokenSource.Cancel();
-                _cancellationTokenSource.Dispose();
```

Comment:
❓ why do we need to dispose of the CTS here? Cancel being called should be sufficient, and it can be cleaned up with garbage collection as needed while safely passing a cancelled token around.

Reply:
It's a good question.  GEnerally speaking, it means:

1. we're a good cleanup citizen.  We're not holding onto things longer than necessary.
2.  we're also representing an invariant that this type wants to convey.  when the ref count of this goes down, the CTS should be entirely finished with that at that point, and any accesses from that point on of _cts.Token *should* throw because it *is* a bug.  This helps us have stronger invariants, which is always preferred in potentially racey code.   

If we don't have this we actually risk doing bad things later and *not* catching it, would could lead to much harder to find issues.